_gitbook/documenting_code/README.md

```diff
@@ -1,6 +1,112 @@
 # Documenting code
 
-Crystal documentation comments use a subset of [Markdown](https://daringfireball.net/projects/markdown/). For example:
+Crystal documentation comments use a subset of [Markdown](https://daringfireball.net/projects/markdown/). 
+
+### Instructions
+
+Documents should be positioned right above definitions of classes, modules, and methods. Leave no blanks between them.
+
+```crystal
+# A unicorn is a **legendary animal** (see the `Legendary` module) that has been
+# described since antiquity as a beast with a large, spiraling horn projecting
+# from its forhead.
+class Unicorn
+end
+
+# Bad: This comment is not attached to any class.
+ 
+class Legendary
+end
+```
+
+* Methods' documentation include Method Summary and Method Detail, while the former includes only the first line of the comments and the latter includes the entire paragraph. In short, a good documentation for a method is to:
+
+  1. State its purpose or functionality in the first line.
+  2. Supplement with details and usages after that.
+
+For instance:
+
+```crystal
+# Returns the number of horns this unicorn has
+#
+# ```
+# Unicorn.new.horns #=> 1
+# ```
+def horns
+  @horns
+end
+```
+
+* Use the third person: `Returns the number` instead of `Return the number`.
+
+* Parameter names should be *italicized* (surrounded with single asterisks `*` or underscores `_`):
+
+```crystal
+# Creates a unicorn with the specified number of *horns*.
+def initialize(@horns = 1)
+  raise "Not a unicorn" if @horns != 1
+end
+```
+
+* Code blocks that have Crystal code can be surrounded with triple backticks or indented with four spaces.
+
+```crystal
+# ```
+# unicorn = Unicorn.new
+# unicorn.speak
+# ```
+```
+
+or
+
+```crystal
+#     unicorn = Unicorn.new
+#     unicorn.speak 
+```
+
+* Text blocks, for example to show program output, must be surrounded with triple backticks followed by the "text" word.
+
+```crystal
+# ```text
+# "I'm a unicorn"
+# ```
+```
+
+* To automatically link to other types, enclose them with single backticks.
+
+```crystal
+# the `Legendary` module
+```
+
+* To automaitcally link to methods of the currently documented type, use a hash, like `#horns` or `#index(char)`.
+
+* To automatically link to methods in other types, do `OtherType#method(arg1, arg2)` or just `OtherType#method`.
+
+* To show expression values inside code blocks, use `#=>`.
+
+```crystal
+1 + 2 #=> 3
+Unicorn.new.speak #=> "I'm a unicorn"
+```
+
+* Use `ditto` to use the same comment as in the previous declaration.
+
+```crystal
+# ditto
+def number_of_horns
+  horns
+end
+```
+
+* Use `:nodoc:` to hide public declarations from the generated documentation. Private and protected methods are always hidden.
+
+```crystal
+# :nodoc:
```

Comment:
This ":nodoc:" is redundant. The example should use ":nodoc:" with something public.